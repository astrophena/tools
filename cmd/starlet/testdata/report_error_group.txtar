-- bot.star --
def send_message(to, text, reply_markup = {}):
    return telegram.call(
        method = "sendMessage",
        args = {
            "chat_id": to,
            "text": text,
            "reply_markup": reply_markup
        }
    )

def handle(update):
    # Invalid arguments, should cause and report an error.
    send_message(config.owner_id)
-- update.json --
{
  "update_id": 10000,
  "message": {
    "date": 1441645532,
    "chat": {
      "title": "foo",
      "id": -100,
      "type": "group"
    },
    "message_id": 1365,
    "from": {
      "last_name": "Test Lastname",
      "id": 123456789,
      "first_name": "Test Firstname",
      "username": "Testusername"
    },
    "forward_from": {
      "last_name": "Forward Lastname",
      "id": 222222,
      "first_name": "Forward Firstname"
    },
    "forward_date": 1441645550,
    "text": "/start"
  }
}
