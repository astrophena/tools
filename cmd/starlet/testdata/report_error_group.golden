[
  {
    "method": "sendMessage",
    "args": {
      "chat_id": "-100",
      "link_preview_options": {
        "is_disabled": true
      },
      "parse_mode": "HTML",
      "text": "‚ùå Something went wrong:\n\u003cpre\u003e\u003ccode\u003eTraceback (most recent call last):\n  bot.star:13:17: in handle\n  bot.star:2:12: in send_message\nError: function send_message missing 1 argument (text)\u003c/code\u003e\u003c/pre\u003e"
    }
  }
]